woocommerce/woocommerce-ios
# Edit Products > price settings: set currency position based on store settings

Fixes #1956

Thanks for the suggestions @pmusolino @shiki ! after a series of unsuccessful attempts to extract the price value, validate, and then reformat the price value with the currency symbol, I tried the approach suggested by @pmusolino to update the currency/price stack view to separate the price input handling from the currency symbol, and it seems to work! (with a small gotcha at the end of the "Changes" section below 👇 )

## Changes

- Added `UIStackView` extension for removing all subviews and adding an array of subviews + basic unit tests
- Added `UnitPosition` enum to `UnitInputViewModel` that indicates where the unit is positioned relative to the input text field
- For `UnitInputViewModel`, passed the `UnitPosition` for the following Product fields:
  - Price settings - regular/sale price: set `UnitPosition` based on wp-admin > WooCommerce > Settings > General
  - Inventory settings - stock quantity: set `UnitPosition` to `none`
  - Shipping settings - height, width, length, weight: set `UnitPosition` to `afterInput`
- In `UnitInputTableViewCell`, updated its stack view's spacing and its subviews based on `UnitPosition`:
  - Unfortunately, I had to set `inputTextField.adjustsFontSizeToFitWidth = false` to turn off automatic font size adjustment to fit width, otherwise the font size shrinks by default probably because the text field inside a `UIStackView` doesn't know its available width. Lemme know if you have any ideas to re-enable this!

## Testing

- Go to the Products tab
- Tap on a simple Product
- Tap on the price settings row --> the currency symbol should be positioned relative to the price based on its store settings under wp-admin > WooCommerce > Settings > General

## Example screenshots/screencast

Screencast (including dynamic type):

![simulator](https://user-images.githubusercontent.com/1945542/77039547-21017580-69f1-11ea-8b94-3ecc5bd5c151.gif)

Under the currency position option in wp-admin > WooCommerce > Settings > General:

<img width="536" alt="Screen Shot 2020-03-19 at 2 57 15 PM" src="https://user-images.githubusercontent.com/1945542/77039971-fcf26400-69f1-11ea-8fac-18d9e9a26386.png">

left with space | left | right with space | right
-- | -- | -- | --
![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 14 45 46](https://user-images.githubusercontent.com/1945542/77039535-1cd55800-69f1-11ea-9569-754c0d1bfba5.png) | ![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 14 46 42](https://user-images.githubusercontent.com/1945542/77039542-1fd04880-69f1-11ea-90bf-20f9ed4e5355.png) | ![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 14 47 09](https://user-images.githubusercontent.com/1945542/77039545-2068df00-69f1-11ea-9bde-533730baf290.png) | ![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 14 47 42](https://user-images.githubusercontent.com/1945542/77039546-21017580-69f1-11ea-90a6-3964229ccb13.png)

sanity tests on other settings screens that have related code changes:

shipping | inventory
-- | --
![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 15 07 46](https://user-images.githubusercontent.com/1945542/77042467-a76c8600-69f6-11ea-905e-1454a1633fcf.png) | ![Simulator Screen Shot - iPhone 11 - 2020-03-19 at 15 07 50](https://user-images.githubusercontent.com/1945542/77041519-04673c80-69f5-11ea-982f-24eff223e7ce.png)

Update release notes:
